test/db/asm/x86_32

```diff
@@ -2156,9 +2162,9 @@ a "xchg eax, edi" 97
 a "xchg eax, edx" 92
 a "xchg eax, esi" 96
 a "xchg eax, esp" 94
-a "xchg ebx, ecx" 87cb
-a "xchg ecx, ebp" 87e9
-a "xchg ecx, ebx" 87d9
+a "xchg ebx, ecx" 87d9
+a "xchg ecx, ebp" 87cd
+a "xchg ecx, ebx" 87cb
```

Comment:
Although the original output was not incorrect per Intel's Instruction Set Reference (`xchg r32, r32` can be interpreted either as an instance of `xchg r32, r/m32` or as `xchg r/m32, r32`), I fixed these lines and the corresponding code to make it in line with the output of other assemblers (NASM and FASM).

Reply:
Mmmm they do seem like different instructions to me. Why the change? Wasn't it working already before?

Reply:
The primary reason for the change is to make the resulting machine code the same as the one produced by other assemblers, namely, NASM and FASM.

Technically, the original output was not incorrect as I stated. It is about which to choose when there are two options. Previously, when encoding `xchg r32, r32`, rasm2 chose the encoding of `xchg r/m32, r32`, whose `Op/En` is `MR`, which means that the left register operand is encoded in the `r/m` part of `ModR/M` and the right in the `reg` part. However, NASM and FASM choose `xchg r32, r/m32` when encoding `xchg r32, r32`. I thought it was a good idea to follow the assemblers widely in use.

The code making the choice is as follows. Flipping 0 and 1 would undo the change, which I can do if desired.

https://github.com/atodekangae/radare2/blob/2f5b8f1d672de7f1933c865293b51e0aac83222d/libr/asm/p/asm_x86_nz.c#L2791-2792

Reply:
@radare opinions?

Reply:
uhm the swap of arguments in test and exchagne instructions on x86 assemnblers and disassemblers have been always a known issue. the thing is.. this change seems to return a different output than capstone, 
``
$ rasm2 -a x86  -d 87cd
xchg ebp, ecx
```